aquadoggo/src/db/stores/document.rs

```diff
@@ -459,6 +459,7 @@ async fn insert_document_view(
             )
         VALUES
             ($1, $2, $3)
+        ON CONFLICT(document_view_id) DO NOTHING -- @TODO: temp fix for double document view insertions: https://github.com/p2panda/aquadoggo/issues/398
```

Comment:
Interesting, could that already be a fix? What are the implications of this change?

Reply:
Yeh, I like it, can't see why not ;-p

Reply:
> Yeh, I like it, can't see why not ;-p

Haha, rock'n'roll.

But also, I'd like really to understand what this change means 😆

Reply:
> I think the only implications are in terms of potential errors being hidden away, but in this case, I don't think there are bad pitfalls 

Can you give some examples?

Reply:
Well, like now, because of this restriction being in place, we know two tasks can exist in the queue which are both resolving the same document view. Guess this isn't a bug, but it tells us about a behaviour we might want to account for. 

The document view row will never change once inserted, so we shouldn't be operating on it again, and we shouldn't be inserting it again. Keeping it as it was before tells us if do either of these things by accident.

Reply:
> The document view row will never change once inserted

If thats the case, Why are even two document views created? Maybe the issue is actually in one of the worker logics?

Reply:
Yeh, i think it's a concurrency issue in the task issuing logic, something like:

- document A has a relation to document B at view 1
- reduce task for document A is in the task queue
- dependency task for document B at view 1 is issued
== operation creating document B at view 1 arrives on the node ==
- reduce task for document B is issued

now two tasks will resolve document B to view 1 and insert it into the db.

Reply:
Agreed that the SQL constraint is very strict here and likely not needed, but I think we should only have it when a) it gives us performance benefits (less checks required, better concurrency etc.) and b) we understand that there is not another bug hiding somewhere.

I need to read through the `reduce` and `dependency` worker code (as you said its been running when the SQL UNIQUE error came up). We might actually deal with a bug there or even worse with the de-duplication logic (though, less likely).

Reply:
Yeh, very happy to figure out if we are confident the worker logic is correct before settling on anything here.

Reply:
> Yeh, i think it's a concurrency issue in the task issuing logic, something like:
> 
>     * document A has a relation to document B at view 1
> 
>     * reduce task for document A is in the task queue
> 
>     * dependency task for document B at view 1 is issued
>       == operation creating document B at view 1 arrives on the node ==
> 
>     * reduce task for document B is issued
> 
> 
> now two tasks will resolve document B to view 1 and insert it into the db.

Aha, so the problem is that we have two concurrent reduce tasks which are considered _different_ since one of them addresses the document via its _id_, and the other via its _view id_?

- `reduce(<abcd ... document id>, None)`
- `reduce(None, <abcd ... document view id>)`

Reply:
Need to think more about this, doesn't sound like an concurrency issue too me either right now (the case I describe above can also happen in series). Maybe the we need to have an upsert instead (either in the `reduce` task or on SQL level as this fix already does).

Reply:
I'll mark it as "solved" here, since we will deal with this in a separate PR